### PR TITLE
#2621 - Convert CP of singletons to singleton

### LINKED ISSUE #2621: Convert CP of singletons to singleton
```julia
julia> x = Singleton(rand(2)); y = Singleton(rand(2));

julia> x × y
CartesianProduct{Float64,Singleton{Float64,Array{Float64,1}},Singleton{Float64,Array{Float64,1}}}(Singleton{Float64,Array{Float64,1}}([0.4945387658006861, 0.9627054076798292]), Singleton{Float64,Array{Float64,1}}([0.051949171335364364, 0.028469563195360648]))

julia> convert(Singleton, x × y)
ERROR: MethodError: Cannot `convert` an object of type 
  CartesianProduct{Float64,Singleton{Float64,Array{Float64,1}},Singleton{Float64,Array{Float64,1}}} to an object of type 
  Singleton
Closest candidates are:
  convert(::Type{Singleton}, ::Singleton) at /home/mforets/.julia/dev/LazySets/src/convert.jl:8
  convert(::Type{T}, ::T) where T at essentials.jl:171
Stacktrace:
 [1] top-level scope at REPL[4]:1
```

### PR DESCRIPTION
**Pull request by mforets**
Closes #2621.

### CODE REVIEW

**Comment by mforets** [comment]
i inadvertently pushed the 2nd commit to master :/ and it is a protected branch so i can't force push to remove it

**Comment by schillic** [comment]
You can un-protect it, force-push, and then protect it again.

File: test/LazyOperations/CartesianProduct.jl [hunk 1]
```diff
@@ -197,6 +197,11 @@ for N in [Float64, Float32, Rational{Int}]
     @test convert(Zonotope, cp) == Zonotope(N[0, 0, 1, 1], N[1 0 0 0; 0 1 0 0;
                                                              0 0 0 1; 0 0 1 0])
 
+    # cartesian product of singletons
+    S1 = Singleton(N[1, 2, 3])
+    S2 = Singleton(N[4, 5, 6])
+    @test convert(S1 × S2, Singleton) == Singleton(N[1, 2, 3, 4, 5, 6])
```
Comment (schillic) [inline test/LazyOperations/CartesianProduct.jl:203]:
```suggestion
    @test convert(Singleton, S1 × S2) == Singleton(N[1, 2, 3, 4, 5, 6])
```

**Comment by mforets** [comment]
> You can un-protect it (here), force-push, and then protect it again.

done

**Review by schillic — APPROVED** [review]
Not sure what happened to the build. Can you retrigger it (maybe with a force-push)?

**Pull request merged**